src/main/thrift/parquet.thrift

```diff
@@ -257,7 +257,14 @@ struct Statistics {
     */
    1: optional binary max;
    2: optional binary min;
-   /** count of null value in the column */
+   /** 
+    * Count of null values in the column.
+    *
+    * Writers SHOULD always write this field even if it is zero (i.e. no null value)
+    * or the column is not nullable.
```

Comment:
Why should writers write this field if the column isn't nullable? That feels strictly redundant and therefore a waste of memory. If a reader assumes that their code be nulls even though the type is required, I would consider this a bug. There wouldn't even be a way to write nulls into a required column.

(Also, if you disregard this comment, at least change the wording from `not nullable` to `not required`, as that's the correct term w.rt. to the schema (there is no `nullable` field in the schema, just a repetition type).)

Reply:
>  Why should writers write this field if the column isn't nullable?

I agree it's not costless in this section, however most impl uses this. It's not a standard but it's widely used.

> (Also, if you disregard this comment, at least change the wording from not nullable to not required, as that's the correct term w.rt. to the schema (there is no nullable field in the schema, just a repetition type).)

You're right, I'll rechecking this